Libplanet/Net/ActionExecutionState.cs

```diff
@@ -0,0 +1,33 @@
+using System.Security.Cryptography;
+using Libplanet.Blocks;
+
+namespace Libplanet.Net
+{
+    /// <summary>
+    /// Indicates a progress of executing block actions.
+    /// </summary>
+    [Equals]
+    public class ActionExecutionState : PreloadState
+    {
+        /// <summary>
+        /// Total number of blocks to execute in the current batch.
+        /// </summary>
+        public int TotalBlockCount { get; internal set; }
+
+        /// <summary>
+        /// The number of currently executed blocks.
+        /// </summary>
+        public int ExecutedBlockCount { get; internal set; }
+
+        /// <summary>
+        /// The hash digest of the block just executed.
+        /// </summary>
+        public HashDigest<SHA256> ExecutedBlockHash { get; internal set; }
+
+        /// <inheritdoc />
+        public override int CurrentPhase => 4;
+
+        /// <inheritdoc />
+        public override int TotalPhase => 4;
```

Comment:
Why did you declared it overrideable?  Shouldn't it be always 4?

Reply:
Then if `TotalPhase` was always 4, how to make user know phase `3` is the last phase when the process to receive state references and block states was successful?

Reply:
That case actually is not a big deal, because it's simply finished earlier than expected.